src/server/plugins/scan.js

```diff
@@ -45,15 +45,20 @@ module.exports = async (kbnServer, server, config) => {
   });
 
   for (let path of pluginPaths) {
-    let modulePath;
     try {
-      modulePath = require.resolve(path);
-    } catch (e) {
-      warning({ tmpl: 'Skipping non-plugin directory at <%= path %>', path: path });
-      continue;
+      const modulePath = require.resolve(path);
+      await plugins.new(path);
+      debug({ tmpl: 'Found plugin at <%= path %>', path: modulePath });
+    } catch (err) {
```

Comment:
Yeah, that's a very good point. A syntax error or something in index.js would cause it to try loading a plugin from package.json